scripts/daq_utils.py

```diff
@@ -0,0 +1,226 @@
+    sb_script += f"#SBATCH --nodelist={nodelist}" + "\n"
+    daqbatch_output = os.path.join(scripts_dir, DAQBATCH_OUTPUT)
+    sb_script += f"#SBATCH --output={daqbatch_output}" + "\n"
+    sb_script += f"#SBATCH --ntasks=1" + "\n"
+    sb_script += f"unset PYTHONPATH" + "\n"
+    sb_script += f"unset LD_LIBRARY_PATH" + "\n"
+    sb_script += cmd
+    daqbatch_script = os.path.join(scripts_dir, DAQBATCH_SCRIPT)
+    with open(daqbatch_script, "w") as f:
+        f.write(sb_script)
+    call_subprocess('sbatch', daqbatch_script)
+    silentremove(daqbatch_script)
+
+class SbatchManager:
+    def __init__(self, user):
+        self.user = user
+
+
+    def get_job_info(self):
+        # Use squeue to get JobId, Comment, JobName, Status, and Reasons (Node List)
+        format_string = '"%i %k %j %T %R"'
+        lines = call_subprocess(
+                    "squeue", "-u", self.user, "-h", "-o", format_string
+                ).splitlines()
+        job_details = {}
+        for i, job_info in enumerate(lines):
+            cols = job_info.strip('"').split()
+            success = True
+            if len(cols) == 5:
+                job_id, comment, job_name, state, nodelist = cols
+            elif len(cols) > 5:
+                job_id, comment, job_name, state = cols[:4]
+                nodelist = " ".join(cols[5:])
+            else:
+                success = False
+            if success:
+                # Get logfile from job_id
+                scontrol_lines = call_subprocess(
+                    "scontrol", "show", "job", job_id
+                ).splitlines()
+                logfile = ""
+                for scontrol_line in scontrol_lines:
+                    if scontrol_line.find("StdOut") > -1:
+                        scontrol_cols = scontrol_line.split("=")
+                        logfile = scontrol_cols[1]
+
+                job_details[job_name] = {
+                    "job_id": job_id,
+                    "comment": comment,
+                    "state": state,
+                    "nodelist": nodelist,
+                    "logfile": logfile,
+                }
+        return job_details
+
+
+class DaqManager:
+    def __init__(self, opts):
+        self.opts = opts
+        self.verbose = False
+        for o, a in self.opts:
+            if o in ('-v', '--verbose'):
+                self.verbose = True
+        self.hutch = call_subprocess("get_info", "--gethutch")
+        self.user = self.hutch+'opr'
+        self.sbman = SbatchManager(self.user)
+        self.scripts_dir = f'/reg/g/pcds/dist/pds/{self.hutch}/scripts'
+        self.set_cnf_file_and_platform()
+
+    def isdaqbatch(self, quiet=False):
+        if self.hutch in DAQBATCH_HUTCHES:
+            if not quiet: print('true')
+            return True
+        else:
+            if not quiet: print('false')
+            return False
+
+    def isvaliduser(self):
+        if getpass.getuser().find('opr') > -1:
```

Comment:
This will find other users, not just hutch operators, if they happen to have `opr` in their usernames.
There are currently two such usernames.
I think you should instead check for `tmoopr` and `txiopr` explicitly using your `DAQBATCH_HUTCHES` variable.

Reply:
Actually, you create self.user and we basically want to check if a given user is self.user @monarin ?

Reply:
Resolved: check user with DAQBATCH_HUTCHES + 'opr'